core/routing.c

```diff
@@ -1692,6 +1692,48 @@ static int uwsgi_route_condition_startswith(struct wsgi_request *wsgi_req, struc
         return 0;
 }
 
+static int uwsgi_route_condition_ipin(struct wsgi_request *wsgi_req, struct uwsgi_route *ur) {
+#define IP4_LEN (sizeof("255.255.255.255")-1)
+	char ipbuf[IP4_LEN+1] = {}, maskbuf[IP4_LEN+1] = {}, pfxlen = 32;
+	char *slash;
+	in_addr_t ip, net, mask;
+
+        char *semicolon = memchr(ur->subject_str, ';', ur->subject_str_len);
+        if (!semicolon) return 0;
+
+        struct uwsgi_buffer *ub = uwsgi_routing_translate(wsgi_req, ur, NULL, 0, ur->subject_str, semicolon - ur->subject_str);
+        if (!ub || ub->pos > IP4_LEN) return -1;
+
+        struct uwsgi_buffer *ub2 = uwsgi_routing_translate(wsgi_req, ur, NULL, 0, semicolon+1, ur->subject_str_len - ((semicolon+1) - ur->subject_str));
+        if (!ub2 || ub2->pos > IP4_LEN) {
```

Comment:
pay attention here (and before for 'ub') as the case for ub->pos do not clear the buffers